server/src/test/java/org/elasticsearch/index/mapper/NumberFieldTypeTests.java

```diff
@@ -346,6 +346,75 @@ public void testLongRangeQueryWithDecimalParts() {
+        final float min_half_float = -65504;
+        final float max_half_float = 65504;
+        assertEquals(
+            ft.rangeQuery(min_half_float, 10, true, true, null, null, null, MOCK_CONTEXT),
+            ft.rangeQuery(-1e+300, 10, true, true, null, null, null, MOCK_CONTEXT)
+        );
+
+        assertEquals(
+            ft.rangeQuery(min_half_float, 10, true, true, null, null, null, MOCK_CONTEXT),
+            ft.rangeQuery(Float.NEGATIVE_INFINITY, 10, true, true, null, null, null, MOCK_CONTEXT)
+        );
+
+        assertEquals(
+            ft.rangeQuery(10, max_half_float, true, true, null, null, null, MOCK_CONTEXT),
+            ft.rangeQuery(10, 1e+300, true, true, null, null, null, MOCK_CONTEXT)
+        );
+
+        assertEquals(
+            ft.rangeQuery(10, max_half_float, true, true, null, null, null, MOCK_CONTEXT),
+            ft.rangeQuery(10, Float.POSITIVE_INFINITY, true, true, null, null, null, MOCK_CONTEXT)
+        );
+
+        assertEquals(
+            ft.rangeQuery(10, 1e+300, false, false, null, null, null, MOCK_CONTEXT),
+            ft.rangeQuery(10, Float.POSITIVE_INFINITY, false, false, null, null, null, MOCK_CONTEXT)
+        );
+
+        assertEquals(
+            ft.rangeQuery(-1e+300, 10, false, false, null, null, null, MOCK_CONTEXT),
+            ft.rangeQuery(Float.NEGATIVE_INFINITY, 10, false, false, null, null, null, MOCK_CONTEXT)
+        );
+    }
+
+    public void testFloatRangeQueryWithOverflowingBounds() {
+        MappedFieldType ft = new NumberFieldMapper.NumberFieldType("field", NumberType.FLOAT, randomBoolean());
+
+        assertEquals(
+            ft.rangeQuery(-Float.MAX_VALUE, 10.0, true, true, null, null, null, MOCK_CONTEXT),
+            ft.rangeQuery(-1e+300, 10.0, true, true, null, null, null, MOCK_CONTEXT)
+        );
+
+        assertEquals(
+            ft.rangeQuery(-Float.MAX_VALUE, 10.0, true, true, null, null, null, MOCK_CONTEXT),
+            ft.rangeQuery(Float.NEGATIVE_INFINITY, 10.0, true, true, null, null, null, MOCK_CONTEXT)
+        );
+
+        assertEquals(
+            ft.rangeQuery(10, Float.MAX_VALUE, true, true, null, null, null, MOCK_CONTEXT),
+            ft.rangeQuery(10, 1e+300, true, true, null, null, null, MOCK_CONTEXT)
+        );
+
+        assertEquals(
+            ft.rangeQuery(10, Float.MAX_VALUE, true, true, null, null, null, MOCK_CONTEXT),
+            ft.rangeQuery(10, Float.POSITIVE_INFINITY, true, true, null, null, null, MOCK_CONTEXT)
+        );
+
+        assertEquals(
+            ft.rangeQuery(10, 1e+300, false, false, null, null, null, MOCK_CONTEXT),
+            ft.rangeQuery(10, Float.POSITIVE_INFINITY, false, false, null, null, null, MOCK_CONTEXT)
+        );
+
+        assertEquals(
+            ft.rangeQuery(-1e+300, 10, false, false, null, null, null, MOCK_CONTEXT),
+            ft.rangeQuery(Float.NEGATIVE_INFINITY, 10, false, false, null, null, null, MOCK_CONTEXT)
+        );
```

Comment:
Just so I understand it, this is saying `include_upper|lower` actually does nothing?

If if we do `Float.MAX_VALUE` and `include_upper` is false?

Reply:
Okay I added some more assertions that I think it's what you asked about initially:

https://github.com/elastic/elasticsearch/blob/5cf6ea112f36cf1ae38bde2c3a705abaf352a689/server/src/test/java/org/elasticsearch/index/mapper/NumberFieldTypeTests.java#L427-L435

You can see that for `-Float.MAX_VALUE` and `+Float.MAX_VALUE` range queries I used `gte` and `lte`.

So a quick diagram that's using the math notation for closed/open intervals:
- `(-1e+300, +1e+300)` becomes `[-Float.MAX_VALUE, +Float.MAX_VALUE]` - because values like `-1e+300` and - `+1e+300` are out of the range of float and cannot be represented as floats. `(-1e+300, +1e+300)` includes `[-Float.MAX_VALUE, +Float.MAX_VALUE]`

for the same reason:
- `[-1e+300, +1e+300]` becomes `[-Float.MAX_VALUE, +Float.MAX_VALUE]`
- `[-Infinity, +Infinity]` becomes `[-Float.MAX_VALUE, +Float.MAX_VALUE]`
- `(-Infinity, +Infinity)` becomes `[-Float.MAX_VALUE, +Float.MAX_VALUE]`